src/main/scala/com/fortysevendeg/android/sqlite/util/CursorUtils.scala

```diff
@@ -0,0 +1,79 @@
+package com.fortysevendeg.android.sqlite.util
+
+import java.lang.reflect.Method
+import java.sql.{PreparedStatement, ResultSet, Types}
+
+import com.fortysevendeg.android.sqlite
+
+import android.database.Cursor
+
+import scala.annotation.tailrec
+import scala.util.{Failure, Success, Try}
+
+object CursorUtils {
+
+  trait CursorProcessor[T] {
+    def move(t: T): Boolean
+    def close(t: T): Unit
+    def isClosed(t: T): Boolean
+  }
+
+  implicit def `Cursor processor` = new CursorProcessor[Cursor] {
+
+    def move(cursor: Cursor): Boolean = cursor.moveToNext()
+
+    def close(cursor: Cursor): Unit = Try(cursor.close())
+
+    def isClosed(cursor: Cursor): Boolean = cursor.isClosed
+  }
+
+  implicit def `ResultSet processor` = new CursorProcessor[ResultSet] {
+
+    def move(resultSet: ResultSet): Boolean = resultSet.next()
+
+    def close(resultSet: ResultSet): Unit = Try(resultSet.close())
+
+    def isClosed(resultSet: ResultSet): Boolean = resultSet.isClosed
+
+  }
+
+  private def processCursor[T, C](c: C)(process: C => T, until: Option[Int] = None)(implicit cursorProcessor: CursorProcessor[C]) = {
+    def untilTrue(size: Int) = until map (_ > size) getOrElse true
+    @tailrec
+    def loop(seq: Seq[T], c: C): Seq[T] = {
+      if (untilTrue(seq.size) && cursorProcessor.move(c)) loop(seq :+ process(c), c)
+      else {
+        cursorProcessor.close(c)
+        seq
+      }
+    }
+    Try(loop(Seq.empty, c)) match {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Default case not specified.](https://www.codacy.com/app/47deg/scala-sqlite-droid/file/784801324/issues/source?bid=2442814&fileBranchId=2493741#l50)